### PR TITLE
Microsoft Defender for Cloud enablement rules - #1632

### PR DESCRIPTION
**Pull request by jdewisscher**
## PR Summary

Check each Microsoft Defender for Cloud Pricing plan is Standard

## PR Checklist

- [ ] PR has a meaningful title
- [ ] Summarized changes
- [ ] Change is not breaking
- [ ] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [ ] Unit tests created/ updated
  - [ ] Rule documentation created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: src/PSRule.Rules.Azure/rules/Azure.Defender.Rule.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+spec:
+  level: Error
+  type:
+  - Microsoft.Security/pricings
+  condition:
+    allOf:
+      - name: '.'
+        equals: 'Containers'
+      - field: 'properties.pricingTier'
+        equals: 'Standard'
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.Defender.Rule.yaml:29]: @jdewisscher Nice. You could make this even better by moving the name check to the pre-condition.

The main difference is:

- If the name check is in the main `condition` block, the rule will fail when other pricing plans exist such as `SqlServers`.
- If you put it in the `where` block, the rule will be skipped for other pricing plans, and will run for `Containers` pricing plans. When run:
  - If the `properties.pricingTier` property is `Standard` the rule will pass.
  - If the property does not exist the rule will fail.
  - If the property is any other value except `Standard` the rule will fail.

For example:

```suggestion
  where:
    name: '.'
    equals: Containers
  type:
  - Microsoft.Security/pricings
  condition:
    allOf:
      - field: 'properties.pricingTier'
        equals: 'Standard'
```

File: tests/PSRule.Rules.Azure.Tests/Azure.Defender.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Fail' });
+            $ruleResult | Should -Not -BeNullOrEmpty;
+            $ruleResult.Length | Should -Be 1;
+            $ruleResult.TargetObject.Name | Should -BeIn 'defenderB';
+
+            # Pass
+            $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Pass' });
+            $ruleResult | Should -Not -BeNullOrEmpty;
+            $ruleResult.Length | Should -Be 1;
+            $ruleResult.TargetObject.Name | Should -BeIn 'defenderA';
```
Comment (BernieWhite) [inline tests/PSRule.Rules.Azure.Tests/Azure.Defender.Tests.ps1:52]: Tests look good.

File: docs/en/rules/Azure.Defender.Containers.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+## NOTES
+
+This rule applies when analyzing resources deployed (in-flight) to Azure.
+
+## LINKS
+
+- [Monitor Azure resources in Microsoft Defender for Cloud](https://docs.microsoft.com/azure/architecture/framework/security/monitor-resources#containers)
+- [Introduction to Microsoft Defender for Containers](https://docs.microsoft.com/en-us/azure/defender-for-cloud/defender-for-containers-introduction)
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.Containers.md:100]:
```suggestion
- [Introduction to Microsoft Defender for Containers](https://docs.microsoft.com/azure/defender-for-cloud/defender-for-containers-introduction)
```

File: src/PSRule.Rules.Azure/rules/Azure.Defender.Rule.yaml (hunk 1 above)
Comment (jdewisscher) [reply src/PSRule.Rules.Azure/rules/Azure.Defender.Rule.yaml]: Excellent point, added!

File: docs/en/rules/Azure.Defender.Containers.md (hunk 3 above)
Comment (jdewisscher) [reply docs/en/rules/Azure.Defender.Containers.md]: Fixed, thank you!

File: docs/en/rules/Azure.Defender.Servers.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+## NOTES
+
+This rule applies when analyzing resources deployed (in-flight) to Azure.
+
+## LINKS
+
+- [Monitor Azure resources in Microsoft Defender for Cloud](https://docs.microsoft.com/azure/architecture/framework/security/monitor-resources#virtual-machines)
+- [Introduction to Microsoft Defender for Containers](https://docs.microsoft.com/azure/defender-for-cloud/defender-for-servers-introduction)
+- [Azure Monitor agent auto-provisioning](https://docs.microsoft.comkj/azure/defender-for-cloud/auto-deploy-azure-monitoring-agent)
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.Servers.md:92]:
```suggestion
- [Azure Monitor agent auto-provisioning](https://docs.microsoft.com/azure/defender-for-cloud/auto-deploy-azure-monitoring-agent)
```

File: docs/en/rules/Azure.Defender.AppServices.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
+---
+severity: Critical
+pillar: Security
+category: Data protection
+resource: Microsoft Defender for App Service
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.AppServices.md:5]:
```suggestion
resource: Microsoft Defender for Cloud
```

File: docs/en/rules/Azure.Defender.Containers.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
+---
+severity: Critical
+pillar: Security
+category: Data protection
+resource: Microsoft Defender for Containers
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.Containers.md:5]:
```suggestion
resource: Microsoft Defender for Cloud
```

File: docs/en/rules/Azure.Defender.SQL.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
+---
+severity: Critical
+pillar: Security
+category: Security operations
+resource: SQL Database
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.SQL.md:5]:
```suggestion
resource: Microsoft Defender for Cloud
```

File: docs/en/rules/Azure.Defender.SQLOnVM.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
+---
+severity: Critical
+pillar: Security
+category: Security operations
+resource: SQL Servers on machines
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.SQLOnVM.md:5]:
```suggestion
resource: Microsoft Defender for Cloud
```

File: docs/en/rules/Azure.Defender.Servers.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
+---
+severity: Critical
+pillar: Security
+category: Virtual Machine
+resource: Microsoft Defender for Servers
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.Servers.md:5]:
```suggestion
resource: Microsoft Defender for Cloud
```

File: docs/en/rules/Azure.Defender.Storage.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
+---
+severity: Critical
+pillar: Security
+category: Data protection
+resource: Microsoft Defender for Storage
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.Storage.md:5]:
```suggestion
resource: Microsoft Defender for Cloud
```

**Review by BernieWhite — APPROVED** [review]
@jdewisscher Awesome work. Just a few suggestions on the documentation to fix a typo and align the naming of the resources to `Microsoft Defender for Cloud`.

Otherwise all good to merge.

**Comment by BernieWhite** [comment]
@jdewisscher Also don't forget to add an entry in the change logs for all the great work. https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md

**Pull request merged**